pkg/apis/v1beta1/nodepool.go

```diff
@@ -160,10 +160,27 @@ type NodePoolList struct {
 	Items           []NodePool `json:"items"`
 }
 
-// OrderByWeight orders the provisioners in the NodePoolList
+// OrderByWeight orders the NodePools in the NodePoolList
 // by their priority weight in-place
+// This priority evaluates the following things in precedence order:
+//  1. NodePools that have a larger weight are ordered first
+//  2. If two NodePools have the same weight, a NodePool that is actually a true NodePool on the apiserver
+//     and isn't a converted NodePool from a Provisioner will come next
```

Comment:
Just a clarifying nit. When I first read this, I thought it could come off as "If i've used the karpenter-convert tool, this means that nodepool would not be prioritized"
```suggestion
//  2. If two NodePools have the same weight, a NodePool that is actually a true NodePool on the apiserver
//     and isn't a converted NodePool from a Provisioner will come next in memory
```

Reply:
Updated wording to be more explicit about the `isProvisioner` flag. Let me know if it's still not clear.